apis/database/v1beta1/nosql_types.go

```diff
@@ -0,0 +1,377 @@
+	MaxStalenessPrefix *int64 `json:"maxStalenessPrefix,omitempty"`
+	// MaxIntervalInSeconds - When used with the Bounded Staleness consistency level, this value represents the time amount of staleness (in seconds) tolerated. Accepted range for this value is 5 - 86400. Required when defaultConsistencyPolicy is set to 'BoundedStaleness'.
+	MaxIntervalInSeconds *int32 `json:"maxIntervalInSeconds,omitempty"`
+}
+
+func toDatabaseLocations(a []Location) *[]documentdb.Location {
+	if a == nil {
+		return &[]documentdb.Location{}
+	}
+
+	s := make([]documentdb.Location, len(a))
+	for i, location := range a {
+		s[i] = documentdb.Location{
+			LocationName:     &location.LocationName,
+			FailoverPriority: &location.FailoverPriority,
+			IsZoneRedundant:  &location.IsZoneRedundant,
+		}
+	}
+
+	return &s
+}
+
+func fromDatabaseLocations(a *[]documentdb.Location) []Location {
+	lenA := 0
+	if a != nil {
+		lenA = len(*a)
+	}
+	s := make([]Location, lenA)
+	if lenA > 0 {
+		for i, location := range *a {
+			s[i] = Location{
+				LocationName:     *location.LocationName,
+				FailoverPriority: *location.FailoverPriority,
+				IsZoneRedundant:  *location.IsZoneRedundant,
+			}
+		}
+	}
+	return s
+}
+
+func sortLocations(a []Location) {
+	sort.SliceStable(a, func(i, j int) bool {
+		return a[i].LocationName < a[j].LocationName
+	})
+}
+
+func checkEqualLocations(a, b []Location) bool {
+	sortLocations(a)
+	sortLocations(b)
+
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := range a {
+		if a[i].LocationName != b[i].LocationName {
+			return false
+		}
+		if a[i].FailoverPriority != b[i].FailoverPriority {
+			return false
+		}
+		if a[i].IsZoneRedundant != b[i].IsZoneRedundant {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Location a region in which the Azure Cosmos DB database account is deployed.
+type Location struct {
+	// LocationName - The name of the region.
+	LocationName string `json:"locationName"`
+	// FailoverPriority - The failover priority of the region. A failover priority of 0 indicates a write region. The maximum value for a failover priority = (total number of regions - 1). Failover priority values must be unique for each of the regions in which the database account exists.
+	FailoverPriority int32 `json:"failoverPriority"`
+	// IsZoneRedundant - Flag to indicate whether or not this region is an AvailabilityZone region
+	IsZoneRedundant bool `json:"isZoneRedundant"`
+}
+
+// A CosmosDBAccountSpec defines the desired state of a CosmosDB Account.
+type CosmosDBAccountSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  CosmosDBAccountParameters `json:"forProvider"`
+}
+
+// An CosmosDBAccountStatus represents the observed state of an Account.
+type CosmosDBAccountStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	State                          string                     `json:"state"`
+	AtProvider                     *CosmosDBAccountParameters `json:"atProvider,omitempty"`
```

Comment:
`State` does not have to be separate from `AtProvider` here and we typically do not duplicate field in both the spec and status of a resource. It looks like it may make sense to have a `CosmosDBAccountObservation` struct that is used for `AtProvider` here and contains `State`